shared/menubar/files-container.desktop.js

```diff
@@ -1,50 +1,32 @@
 // @flow
+import * as React from 'react'
 import * as FsTypes from '../constants/types/fs'
 import * as FsGen from '../actions/fs-gen'
 import * as FsUtil from '../util/kbfs'
 import * as FsConstants from '../constants/fs'
 import * as TimestampUtil from '../util/timestamp'
-import {FilesPreview} from './files.desktop'
+import {FilesPreview, type UserTlfUpdateRowProps} from './files.desktop'
 import {remoteConnect, compose} from '../util/container'
 
 const mapStateToProps = (state) => ({
   _username: state.username,
-  _userTlfUpdates: [
-    {
-      tlf: FsTypes.stringToPath('/keybase/team/zila.test'),
-      writer: 'jzila',
-      timestamp: 1535497273,
-      updates: [
-        FsTypes.stringToPath('/keybase/team/zila.test/abc'),
-      ],
-    },
-    {
-      tlf: FsTypes.stringToPath('/keybase/team/zila.test'),
-      writer: 'songgao',
-      timestamp: 1535497273,
-      updates: [
-        FsTypes.stringToPath('/keybase/team/zila.test/ghi.txt'),
-        FsTypes.stringToPath('/keybase/team/zila.test/def'),
-        FsTypes.stringToPath('/keybase/team/zila.test/abc'),
-      ],
-      moreUpdateCount: 17,
-    },
-  ],
+  _userTlfUpdates: state.fileRows,
 })
 
 const mapDispatchToProps = dispatch => ({
   _onSelectPath: (path: FsTypes.Path) => dispatch(FsGen.createOpenFilesFromWidget({path})),
-  onViewAll: () => dispatch(FsGen.createOpenFilesFromWidget({})),
+  loadTlfUpdates: () => dispatch(FsGen.createUserFileEditsLoad()),
 })
 
 const mergeProps = (stateProps, dispatchProps) => ({
-  onViewAll: dispatchProps.onViewAll,
+  loadTlfUpdates: dispatchProps.loadTlfUpdates,
   userTlfUpdates: stateProps._userTlfUpdates.map(c => {
-    const {participants, teamname} = FsUtil.tlfToParticipantsOrTeamname(FsTypes.pathToString(c.tlf))
+    const tlf = FsTypes.pathToString(c.tlf)
+    const {participants, teamname} = FsUtil.tlfToParticipantsOrTeamname(tlf)
     const iconSpec = FsConstants.getIconSpecFromUsernamesAndTeamname([c.writer], null, stateProps._username)
     return {
       onSelectPath: () => dispatchProps._onSelectPath(c.tlf),
-      tlf: FsTypes.pathToString(c.tlf),
+      tlf,
```

Comment:
Why not just pass the original `tlf` to component?

Reply:
Because the component needs it as a string.

Reply:
What I was trying to ask was why doesn't component just take a `Types.Path` instead.